pdns/lua-record.cc

```diff
@@ -1047,6 +1047,16 @@ static void setupLuaRecords(LuaContext& lua) // NOLINT(readability-function-cogn
         string dashed=ip6.toString();
         boost::replace_all(dashed, ":", "-");
 
+        // https://github.com/PowerDNS/pdns/issues/7524
+        if (boost::ends_with(dashed, "-")) {
+          // "a--a-" -> "a--a-0"
+          dashed.push_back('0');
+        }
+        if (boost::starts_with(dashed, "-") || dashed.compare(2, 2, "--") == 0) {
```

Comment:
Isn't there a risk of `compare` throwing an exception if `dashed` is smaller than 2 characters?

Reply:
I -believe- the shortest possible string we'll have at this point is long enough

Reply:
Probably safe enough as the exception will be caught anyway.

Reply:
yes - and we know that that will "neatly" cause `SERVFAIL`